packages/spatial/src/renderer/components/InfiniteGridHelper.ts

```diff
@@ -134,102 +128,142 @@ void main() {
 }
 `
 
-export class InfiniteGridHelper extends Mesh<PlaneGeometry, ShaderMaterial> {
-  plane: Plane
+export const InfiniteGridHelperComponent = defineComponent({
+  name: 'InfiniteGridHelperComponent',
 
-  constructor(size1 = 1, size2 = 10, color = new Color('white'), distance = 8000) {
-    const geometry = new PlaneGeometry(2, 2, 1, 1)
-    const material = new ShaderMaterial({
-      side: DoubleSide,
-      uniforms: {
-        uSize1: {
-          value: size1
-        },
-        uSize2: {
-          value: size2
-        },
-        uColor: {
-          value: color
+  onInit: (entity) => {
+    return {
+      name: 'infinite-grid-helper',
+      size1: 1,
+      size2: 10,
+      color: new Color('white') as ColorRepresentation,
+      distance: 8000,
+      gridHeight: 0
+    }
+  },
+
+  onSet: (entity, component, json) => {
+    if (!json) return
+
+    if (typeof json.name == 'string') component.name.set(json.name)
+    if (typeof json.size1 == 'number') component.size1.set(json.size1)
+    if (typeof json.size2 == 'number') component.size2.set(json.size2)
+    // if (matchesColor.test(json.color)) component.color.set(json.color)
+    if (typeof json.distance == 'number') component.distance.set(json.distance)
+    if (typeof json.gridHeight == 'number') component.gridHeight.set(json.gridHeight)
+  },
+
+  reactor: function () {
+    const entity = useEntityContext()
+    const component = useComponent(entity, InfiniteGridHelperComponent)
+    const [mesh, geometry, material] = useMeshComponent(
+      entity,
+      new PlaneGeometry(2, 2, 1, 1),
```

Comment:
again, this will create new objects every render. though now that i think about it, it might just be a garbage inefficiency if class creation doesn't result in any side effects inside of three

Reply:
Ya, I implemented this one incorrectly at the API level, I'll update it. The other places that were creating objects weren't updated to use the correct overload of useResource, those are fixed.